dallinger/recruiters.py

```diff
@@ -381,6 +487,68 @@ def _handle_exit_form_submission(self, assignment_id: str, participant_id: str):
         q = _get_queue()
         q.enqueue(worker_function, "AssignmentSubmitted", assignment_id, participant_id)
 
+    def load_service(self, sandbox):
+        return _prolific_service_from_config(sandbox)
+
+    def clean_qualifications(self, experiment_details):
+        requirements = []
+        for requirement in experiment_details["eligibility_requirements"]:
+            cleaned_attributes = []
+            for attribute in requirement["attributes"]:
+                if (
+                    attribute["value"] is False
+                    or attribute["value"] is None
+                    or attribute["value"] == []
+                ):
+                    continue
+
+                elif (
+                    requirement["type"] == "input"
+                    and attribute["value"] == attribute["default_value"]
+                ):
+                    continue
+                cleaned_attributes.append(attribute)
+            attributes = requirement["attributes"]
+            if requirement["type"] == "range":
+                if len(attributes) == 0:
+                    continue
+                if (
+                    attributes[0]["min"] == attributes[0]["value"]
+                    and attributes[1]["max"] == attributes[1]["value"]
+                ):
+                    continue
+
+            requirement["attributes"] = cleaned_attributes
+            if len(cleaned_attributes) > 0:
+                try:
+                    query_id = requirement["query"]["id"]
```

Comment:
Refactor into a function called `clean_query`

Reply:
Document this function to explain why the queries need cleaning

Reply:
I refactored it into a separate function `clean_query` and documented why the queries need cleaning.

Reply:
Nice